src/account-mgr.cpp

```diff
@@ -207,3 +204,41 @@ Account AccountManager::getAccountBySignature(const QString& account_sig) const
 
     return Account();
 }
+
+void AccountManager::updateServerInfo()
+{
+    for (size_t i = 0; i < accounts_.size(); i++)
+        updateServerInfo(i);
+}
+
+void AccountManager::updateServerInfo(unsigned index)
+{
+    ServerInfoRequest *request;
+    // request is taken owner by Account object
+    request = accounts_[index].createServerInfoRequest();
+    connect(request, SIGNAL(success(const Account&, const ServerInfo &)),
+            this, SLOT(serverInfoSuccess(const Account&, const ServerInfo &)));
+    connect(request, SIGNAL(failed(const ApiError&)),
+            this, SLOT(serverInfoFailed(const ApiError&)));
+    request->send();
+}
+
+
+void AccountManager::serverInfoSuccess(const Account &account, const ServerInfo &info)
+{
+    for (size_t i = 0; i < accounts_.size(); i++) {
+        if (accounts_[i] == account) {
+            if (i == 0)
+                emit beforeAccountChanged();
+            accounts_[i].serverInfo = info;
+            if (i == 0)
+                emit accountsChanged();
```

Comment:
As I mentioned, emitting `accountChanged` signal here would cause unnecessary operations on other modules which connects to this signal. For example, the avatar module would clear the in-memory cache of avatars, and the seahub notification module would send a request immediately to query unread notifications. 

To avoid these, my suggestion is: add another signal `accountInfoUpdated`, and emit this signal instead of `accountChanged`. account-view.cpp should listen to this new signal, so that the other modules would not be affected. The `accountChanged` signal is designed to be used when the user switches to another account.

> But if we use extra signals, it will involve a lot of complexities when you need to detect whether it has pro feature, and add extra code there.

I see your consideration here. But I don't think there are many places in the code where we need to write these possible extra code.